onmt/opts.py

```diff
@@ -511,6 +511,14 @@ def translate_opts(parser):
               help='Source directory for image or audio files')
     group.add('--tgt', '-tgt',
                        help='True target sequence (optional)')
+    group.add('--shard_size', '-shard_size', type=int, default=1000000,
```

Comment:
1 million seems like a very large default shard size. Is not sharding at translation time a better default? 

I just noticed that setting `shard_size=0` does not have the intended effect -- it fails to read the file at all. I'll try to have a fix for this soon.

Reply:
it all depends on the machine you're doing this :) but yes I'll set it to 10k
Yes I also noticed the shard_size=0 not being effective, letting you fix it.